lib/unparser/emitter/primitive.rb

```diff
@@ -20,6 +20,23 @@ def dispatch
 
       end # Inspect
 
+      class Symbol < self
+
+        handle :sym
+
+        private
+
+        # mutant:disable
+        def dispatch
+          if RUBY_VERSION < '3.2' && value[-1] == '='
```

Comment:
@prikha if you had mutation tested this on 3.1, it should have forced you into `value[-1].eql?('=')` which is better code as it uses the non coercing `#==` (even if coercion does not trigger here in the first place, its just less powerful primitive).

But in addition, I do not think we should pull a string out to than compare with a literal. We can AFAIK just directly uses regexps so `value.match?(/=\z/)`. This creates no intermediary objects and avoids also to access out of bounds on empty symbols.

Reply:
Applied